manifests/init.pp

```diff
@@ -5,100 +5,78 @@
-
-  $rhsm_url                     = $foreman_proxy_content::params::rhsm_url,
-
-  $qpid_router                  = $foreman_proxy_content::params::qpid_router,
-  $qpid_router_hub_addr         = $foreman_proxy_content::params::qpid_router_hub_addr,
-  $qpid_router_hub_port         = $foreman_proxy_content::params::qpid_router_hub_port,
-  $qpid_router_agent_addr       = $foreman_proxy_content::params::qpid_router_agent_addr,
-  $qpid_router_agent_port       = $foreman_proxy_content::params::qpid_router_agent_port,
-  $qpid_router_broker_addr      = $foreman_proxy_content::params::qpid_router_broker_addr,
-  $qpid_router_broker_port      = $foreman_proxy_content::params::qpid_router_broker_port,
-  $qpid_router_logging_level    = $foreman_proxy_content::params::qpid_router_logging_level,
-  $qpid_router_logging_path     = $foreman_proxy_content::params::qpid_router_logging_path,
-  $enable_ostree                = $foreman_proxy_content::params::enable_ostree,
+  String $parent_fqdn = $foreman_proxy_content::params::parent_fqdn,
+  Optional[Stdlib::Absolutepath] $certs_tar = $foreman_proxy_content::params::certs_tar,
+  Boolean $pulp_master = $foreman_proxy_content::params::pulp_master,
+  String $pulp_admin_password = $foreman_proxy_content::params::pulp_admin_password,
+  String $pulp_oauth_effective_user = $foreman_proxy_content::params::pulp_oauth_effective_user,
+  String $pulp_oauth_key = $foreman_proxy_content::params::pulp_oauth_key,
+  Optional[String] $pulp_oauth_secret = $foreman_proxy_content::params::pulp_oauth_secret,
+  Optional[String] $pulp_max_speed = $foreman_proxy_content::params::pulp_max_speed,
+
+  Boolean $puppet = $foreman_proxy_content::params::puppet,
+
+  Boolean $reverse_proxy = $foreman_proxy_content::params::reverse_proxy,
+  Integer[0, 65535] $reverse_proxy_port = $foreman_proxy_content::params::reverse_proxy_port,
+
+  String $rhsm_url = $foreman_proxy_content::params::rhsm_url,
+
+  Boolean $qpid_router = $foreman_proxy_content::params::qpid_router,
+  String $qpid_router_hub_addr = $foreman_proxy_content::params::qpid_router_hub_addr,
+  Integer[0, 65535] $qpid_router_hub_port = $foreman_proxy_content::params::qpid_router_hub_port,
+  String $qpid_router_agent_addr = $foreman_proxy_content::params::qpid_router_agent_addr,
+  Integer[0, 65535] $qpid_router_agent_port = $foreman_proxy_content::params::qpid_router_agent_port,
+  String $qpid_router_broker_addr = $foreman_proxy_content::params::qpid_router_broker_addr,
+  Integer[0, 65535] $qpid_router_broker_port = $foreman_proxy_content::params::qpid_router_broker_port,
+  String $qpid_router_logging_level = $foreman_proxy_content::params::qpid_router_logging_level,
+  Stdlib::Absolutepath $qpid_router_logging_path = $foreman_proxy_content::params::qpid_router_logging_path,
+  Boolean $enable_ostree = $foreman_proxy_content::params::enable_ostree,
 ) inherits foreman_proxy_content::params {
   validate_bool($enable_ostree)
```

Comment:
The third can be replaced with ```assert_type(String, $pulp_oauth_secret)```. Maybe ```String[5]``` to set a minimum secret length. That would drop the dependency on puppet-common as well.